doc/sphinx/language/gallina-extensions.rst

```diff
@@ -869,255 +870,297 @@ Module system
 
-    .. cmdv:: Declare Module @ident {* @module_binding} : @module_type
+.. flag:: Short Module Printing
 
-       Declares a functor with parameters given by the list of :token:`module_binding` and output module type
-       :token:`module_type`.
+   This flag (off by default) disables the printing of the types of fields,
+   leaving only their names, for the commands :cmd:`Print Module` and
+   :cmd:`Print Module Type`.
 
-.. example::
+.. _module_examples:
 
-    Let us define a simple module.
+Examples
+~~~~~~~~
 
-    .. coqtop:: all
+.. example:: Defining a simple module interactively
 
-       Module M.
+    .. coqtop:: in
 
+       Module M.
        Definition T := nat.
-
        Definition x := 0.
 
-       Definition y : bool.
+    .. coqtop:: all
 
+       Definition y : bool.
        exact true.
 
-       Defined.
+    .. coqtop:: in
 
+       Defined.
        End M.
 
-Inside a module one can define constants, prove theorems and do any
-other things that can be done in the toplevel. Components of a closed
+Inside a module one can define constants, prove theorems and do anything
+else that can be done in the toplevel. Components of a closed
 module can be accessed using the dot notation:
 
 .. coqtop:: all
 
    Print M.x.
 
-A simple module type:
-
-.. coqtop:: all
-
-   Module Type SIG.
-
-   Parameter T : Set.
+.. _example_def_simple_module_type:
 
-   Parameter x : T.
+.. example:: Defining a simple module type interactively
 
-   End SIG.
+   .. coqtop:: in
 
-Now we can create a new module from M, giving it a less precise
-specification: the y component is dropped as well as the body of x.
+      Module Type SIG.
+      Parameter T : Set.
+      Parameter x : T.
+      End SIG.
 
-.. coqtop:: all
+.. _example_filter_module:
 
-   Module N : SIG with Definition T := nat := M.
+.. example Creating a new module that omits some items from an existing module
 
-   Print N.T.
+   Since the type, :n:`SIG`, of the new module :n:`N` doesn't define y or give the body of :n`x`,
+   those are not included in :n:`N`.
 
-   Print N.x.
+   .. coqtop:: all
 
-   Fail Print N.y.
+      Module N : SIG with Definition T := nat := M.
+      Print N.T.
+      Print N.x.
+      Fail Print N.y.
 
 .. reset to remove N (undo in last coqtop block doesn't seem to do that), invisibly redefine M, SIG
 .. coqtop:: none reset
 
    Module M.
-
    Definition T := nat.
-
    Definition x := 0.
-
    Definition y : bool.
-
    exact true.
-
    Defined.
-
    End M.
 
    Module Type SIG.
-
    Parameter T : Set.
-
    Parameter x : T.
-
    End SIG.
 
-The definition of :g:`N` using the module type expression :g:`SIG` with
+The following definition of :g:`N` using the module type expression :g:`SIG` with
 :g:`Definition T := nat` is equivalent to the following one:
 
-.. coqtop:: all
+.. todo: what is other definition referred to above?
+   "Module N' : SIG with Definition T := nat. End N`." is not it.
```

Comment:
I don't know. These examples will need to be reworked, but maybe not now. Please do leave your `.. todo` note in the code.